### PR TITLE
Sidekiq integration

### PR DESCRIPTION
**Pull request by rossta**
Significant refactoring of `ActiveAsync` internal and external API to match `ActiveJob`. Adds support for `Sidekiq`

- [x] refactor to queue adapters (matching active job api)
- [x] implement sidekiq adapter

### CODE REVIEW

File: activeasync.gemspec [hunk 1]
```diff
@@ -18,13 +18,14 @@ Gem::Specification.new do |s|
   s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
   s.require_paths = ["lib"]
 
-  # specify any dependencies here; for example:
-  # s.add_runtime_dependency "rest-client"
-  s.add_dependency "resque", "~> 1.10"
```
Comment (midu) [inline activeasync.gemspec]: :hellyeah:

File: lib/active_async/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module ActiveAsync
-  VERSION = "0.0.3"
+  VERSION = "0.1.0"
```
Comment (midu) [inline lib/active_async/version.rb]: is 1.0 a version that actually uses activejob?

**Comment by midu** [comment]
I'm not going to pretend I understand 100% of what's happening here, but I like it!

**Comment by rossta** [comment]
Yes, the idea is we can upgrade this independently of Rails 4.2 and remain backward compatible until then. Something like 0.2 would support both, then 0.3+ could be activejob compatible only.

**Comment by MGerrior** [comment]
@rossta Any plans on moving this forward or shall I take over? It looks like we could merge this, add an initialize that says `ActiveAsync.queue_adapter = :sidekiq` and that would help us move forward with dropping Resque support.

**Comment by rossta** [comment]
@MGerrior If you'd like to help me move this forward, that would be great! Nothing keeping this from happening besides effort to integrate and test.

**Comment by MGerrior** [comment]
Passing build, merging this in.

**Pull request merged**